### PR TITLE
flexbox: initial attempt to add support for RTL direction

### PR DESCRIPTION
**Pull request by malnajdi**
I have added:
1. 2 test cases for flexbox with direction RTL. 
2. modified the code which handles flexbox to add RTL aware code.

related #601
related #106

### CODE REVIEW

**Review by liZe — CHANGES_REQUESTED** [review]

File: weasyprint/layout/flex.py [hunk 1]
```diff
@@ -650,33 +650,61 @@ def flex_layout(context, box, max_position_y, skip_stack, containing_block,
...
 
-        if justify_content == 'flex-end':
-            position_axis += free_space
-        elif justify_content == 'center':
-            position_axis += free_space / 2
-        elif justify_content == 'space-around':
-            position_axis += free_space / len(line) / 2
-        elif justify_content == 'space-evenly':
-            position_axis += free_space / (len(line) + 1)
+        if box.style['direction'] == 'rtl':
```
[the remark this answers is not in the document]
Comment (malnajdi) [reply weasyprint/layout/flex.py]: I have added extra if statement to ensure the logic is applied only if  `flex-direction` starts with row-

[the remark this answers is not in the document]
Comment (malnajdi) [reply weasyprint/layout/flex.py]: I have tried to add what you required but it breaks the flex items

File: weasyprint/tests/test_layout/test_flex.py [hunk 2]
```diff
@@ -64,6 +89,35 @@ def test_flex_direction_row_reverse():
...
+        div_3.position_y ==
+        article.position_y)
+    # below assert needs to be fixed
+    assert (
+        div_3.position_x + div_3.width ==
+        article.position_x + article.width)
+    assert div_1.position_x > div_2.position_x > div_3.position_x
+
+
 @assert_no_logs
```
[the remark this answers is not in the document]
Comment (malnajdi) [reply weasyprint/tests/test_layout/test_flex.py:120]: done

File: weasyprint/tests/test_layout/test_flex.py [hunk 3]
```diff
@@ -58,7 +58,9 @@ def test_flex_direction_row_rtl():
         div_2.position_y ==
         div_3.position_y ==
         article.position_y)
-    assert div_1.position_x == article.width - div_1.padding_width()
+    assert (
```
Comment (malnajdi) [inline weasyprint/tests/test_layout/test_flex.py:61]: @liZe please verify if this test is correct.

File: weasyprint/tests/test_layout/test_flex.py [hunk 4]
```diff
@@ -111,10 +113,7 @@ def test_flex_direction_row_reverse_rtl():
         div_2.position_y ==
         div_3.position_y ==
         article.position_y)
-    # below assert needs to be fixed
-    assert (
-        div_3.position_x + div_3.width ==
-        article.position_x + article.width)
+    assert div_3.position_x == article.position_x
```
Comment (malnajdi) [inline weasyprint/tests/test_layout/test_flex.py:116]: @liZe please verify if this test is correct.

**Comment by liZe** [comment]
Thanks a lot! No problem, I’ll simplify the "if rtl" part of your PR in a future commit.

**Pull request merged**